src/glslplugin/lang/elements/GLSLIdentifier.java

```diff
@@ -20,22 +20,50 @@
 package glslplugin.lang.elements;
 
 import com.intellij.lang.ASTNode;
+import com.intellij.psi.PsiCheckedRenameElement;
 import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiNameIdentifierOwner;
 import com.intellij.psi.PsiReference;
+import com.intellij.util.IncorrectOperationException;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-public class GLSLIdentifier extends GLSLElementImpl {
+public class GLSLIdentifier extends GLSLElementImpl implements PsiCheckedRenameElement, PsiNameIdentifierOwner {
 
     public GLSLIdentifier(@NotNull ASTNode astNode) {
         super(astNode);
     }
 
+    @Nullable
+    @Override
+    public PsiElement getNameIdentifier() {
+        return findChildByType(GLSLTokenTypes.IDENTIFIER);
+    }
+
     @NotNull
-    public String getIdentifierName() {
+    public String getName() {
         return getText();
     }
 
+    @Override
+    public void checkSetName(String name) throws IncorrectOperationException {
+        final PsiElement oldName = getNameIdentifier();
+        if (oldName == null) throw new IncorrectOperationException("Unnamed identifier! (This probably shouldn't happen.)");
+        PsiElement newName = GLSLPsiElementFactory.createLeafElement(getProject(), name);
+        if (newName.getNode().getElementType() != GLSLTokenTypes.IDENTIFIER)
+            throw new IncorrectOperationException("Invalid identifier name!");
+    }
+
+    @Override
+    public PsiElement setName(@NotNull String name) throws IncorrectOperationException {
+        checkSetName(name);
+        final PsiElement oldName = getNameIdentifier();
+        assert oldName != null;
```

Comment:
Please replace this assert with something more graceful, sometimes things that can return null, will return null, no matter how unlikely it seems. You can log it though, if you are curious when it will explode.